packages/thumbprint-react/components/Tooltip/index.tsx

```diff
@@ -153,7 +167,27 @@ export default function Tooltip({
         }
     };
 
-    useCloseOnEscape(hide, canUseDOM);
+    const onBlur = (): void => {
+        // By default this adds a small delay before closing which gives time
+        // to focus by keyboard tab inside tooltip content for rich tooltip.
+        setCloseTimeout(window.setTimeout(hide, closeDelayLength));
+        setHasOpenedByOnClick(false);
+    };
+
+    const getAriaLabel = (): string => {
+        if (ariaLabel) {
+            return ariaLabel;
+        }
+        if (typeof text === 'string') {
```

Comment:
I think there's a utility function that can extract text from FormattedText. Might be good here, otherwise any rich text won't get proper treatment if ariaLabel isn't present

Reply:
Yeah below function can be used to extract `ariaLabel` from `FormattedText`.

 _const ariaLabel = text.segments.map((segment) => segment.text).join('');_

But, text can have type of React.DOM / string. I am not sure how to extract **_ariaLabel_** from **_React.DOM_**.

Reply:
technically, you can get it with textContent by setting a ref against the DOM element, but I'm not sure it's worth going through this much trouble. Perhaps ariaLabel can be required if the input is React.DOM?

Reply:
That's a good idea to use ariaLabel prop incase of  input is React.DOM. Let me add comment in ariaLabel prop types to be passed in that case. But, ariaLabel prop type will be non-required as it can be used for string inputs as well.

Reply:
[Done](https://github.com/thumbtack/thumbprint/pull/963/commits/62fdb39a40d34c7c9b59a890da13d95183a7e4e9#diff-e873722ff811ec99525e10534f21cca386d7f8a89cf12299b5601ddb860969a7R100)